app/modules/settings/plugins/index.js

```diff
@@ -56,8 +56,12 @@ function cleanUpConfiguration (schema, configuration) {
     configuration[key] = item
     if (!keepItem(item) || !schema.properties || !(key in schema.properties)) {
       delete configuration[key]
-    } else if (schema.properties && schema.properties[key] && schema.properties[key].type === 'object') {
-      cleanUpConfiguration(schema.properties[key], item)
+    } else if (schema.properties && schema.properties[key]) {
+      if (schema.properties[key].type === 'integer') {
```

Comment:
You should put `schema.properties[key].type` in a local variable to avoid duplication.